packages/jobs/lib/runner/lambda.ts

```diff
@@ -0,0 +1,195 @@
+        }
+        return Lambda.instance;
+    }
+
+    async createFunction(node: Node): Promise<Result<void>> {
+        setImmediate(async () => {
+            const name = getFunctionName(node);
+            try {
+                const createFunctionCommand = new CreateFunctionCommand({
+                    FunctionName: name,
+                    Role: envs.LAMBDA_EXECUTION_ROLE_ARN,
+                    Code: {
+                        ImageUri: `${envs.LAMBDA_ECR_REGISTRY}/${node.image}`
+                    },
+                    PackageType: 'Image',
+                    Architectures: [envs.LAMBDA_ARCHITECTURE],
+                    MemorySize: getSize(node),
+                    Timeout: node.executionTimeoutSecs || envs.LAMBDA_EXECUTION_TIMEOUT_SECS,
+                    Environment: this.getEnvironmentVariables(node),
+                    VpcConfig: {
+                        SubnetIds: envs.LAMBDA_SUBNET_IDS,
+                        SecurityGroupIds: envs.LAMBDA_SECURITY_GROUP_IDS
+                    }
+                });
+                const cfResult = await lambdaClient.send(createFunctionCommand);
+                await waitUntilFunctionActive(
+                    { client: lambdaClient, maxWaitTime: envs.LAMBDA_CREATE_TIMEOUT_SECS },
+                    {
+                        FunctionName: cfResult.FunctionName
+                    }
+                );
+                const publishVersionCommand = new PublishVersionCommand({
+                    FunctionName: cfResult.FunctionName
+                });
+                const pvResult = await lambdaClient.send(publishVersionCommand);
+                await waitUntilPublishedVersionActive(
+                    { client: lambdaClient, maxWaitTime: envs.LAMBDA_CREATE_TIMEOUT_SECS },
+                    {
+                        FunctionName: pvResult.FunctionName,
+                        Qualifier: pvResult.Version
+                    }
+                );
+                const aResult = await lambdaClient.send(
+                    new CreateAliasCommand({
+                        FunctionName: pvResult.FunctionName,
+                        Name: envs.LAMBDA_FUNCTION_ALIAS,
+                        FunctionVersion: pvResult.Version
+                    })
+                );
+                await lambdaClient.send(
+                    new PutProvisionedConcurrencyConfigCommand({
+                        FunctionName: pvResult.FunctionName,
+                        Qualifier: envs.LAMBDA_FUNCTION_ALIAS,
+                        ProvisionedConcurrentExecutions: node.provisionedConcurrency || envs.LAMBDA_PROVISIONED_CONCURRENCY
+                    })
+                );
+                const fleetId = node.fleetId || envs.RUNNER_LAMBDA_FLEET_ID;
+                const result = await registerWithFleet(fleetId, {
+                    nodeId: node.id,
+                    url: `${aResult.AliasArn}`
+                });
```

Comment:
Instance-based runners report their readiness to the `jobs-service` when they have started - this is not inline with how function-based runners will work so the provider itself will register the created function with the fleet when all creation operations have completed.